### PR TITLE
[MINSTALL-187] Switch to the Maven 4 API

### PR DESCRIPTION
**Pull request by gnodet**
JIRA issue: https://issues.apache.org/jira/browse/MINSTALL-187
Requires:
 * https://github.com/apache/maven/pull/1309
 * https://github.com/apache/maven-plugin-testing/pull/39
 * https://github.com/apache/maven-plugin-tools/pull/242

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -169,24 +131,39 @@
...
+      <artifactId>junit-jupiter</artifactId>
+      <version>5.10.0</version>
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.aether</groupId>
-      <artifactId>aether-impl</artifactId>
-      <version>${aetherVersion}</version>
+      <groupId>jakarta.enterprise</groupId>
+      <artifactId>jakarta.enterprise.cdi-api</artifactId>
```
Comment (cstamas) [inline pom.xml:146]: Huh? Why not inject?

Comment (cstamas) [reply pom.xml:146]: cdi-api is a 120k beast, while inject is 10k?

Comment (gnodet) [reply pom.xml]: Removed dep.

**Comment by jira-importer** [comment]
Resolve #290

**Pull request merged**